### ISSUE TITLE
gp open reports http timeout

### ISSUE THREAD

**Issue by stuartpb** [gitpod-io/gitpod#564]
This is a duplicate of #519 that I can reproduce in my own workspaces.

**Describe the bug**

When I run `gp open somefile.txt` the file opens, but `gp` does not return when it's done and reports `error while issuing request: Post http://localhost:23000/gitpod/cli: net/http: request canceled (Client.Timeout exceeded while awaiting headers)` after a while.

**To Reproduce**

Run `gp open somefile.txt`

**Expected behavior**

`gp` should exit as soon as the file is opened (possibly with an argument that allows for blocking until the tab is closed, akin to Cloud9's `c9 open --wait`).

**Observed behavior**

`gp` blocks until the timeout described above.

**Comment by stuartpb** [comment]
Seems to be related to pods that might have not been running, but the state was preserved, which is a thing I've seen.

**Comment by stuartpb** [comment]
Or rather, I think this might have something to do with workspaces that have multiple heads connecting to them? It seems like this is more likely to happen when I've got a couple laptops open to the same workspace.

**Comment by svenefftinge** [comment]
We do the wait by default on purpose, because we want to allow opening files on workspace start through `tasks` in `.gitpod.yml`. But it should return after it opened the editor and not show an error.

**Comment by stuartpb** [comment]
Is there some kind of traceback or dump/log I can provide to diagnose this issue?

**Comment by svenefftinge** [comment]
You can share the workspace id (part of the URL) and we can look if there's anything suspicious in the logs. But I can imagine doing gp open with multiple frontends connected could be the cause. As we are waiting only for the first to open.

**Comment by corneliusludmann** [comment]
This line in my `.gitpod.yml` gives me the same error on every workspace start:

https://github.com/corneliusludmann/hello-android/blob/7810254d23edf570491d1ebb831590f7a1c46b97/.gitpod.yml#L22
```
          gp env "APPETIZE_PUBLICKEY=$APPETIZE_PUBLICKEY"
```

```
2020/06/03 07:50:52 error while issuing request: Post "http://localhost:23000/gitpod/cli": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

However, setting the env variable works as expected. It's just the noisy error message.

**Comment by nisarhassan12** [comment]
I'm trying to await port 4200 here via `gp await-port 4200` https://github.com/nisarhassan12/open-event-frontend/blob/54c3257932d5032e82270efe9b2a7a1f13cd4db7/.gitpod.yml#L2 but it seems to be failing with the following error:

![image](https://user-images.githubusercontent.com/46004116/95125604-a6226100-076e-11eb-9c58-7d4029e3f383.png)

**Comment by nisarhassan12** [comment]
Having a similar issue for https://github.com/nisarhassan12/svgr

https://github.com/TypeFox/gitpodify/issues/476

The port did not seem to be working I was getting port 800 didn't respond which does not usually happen in a Gatsby project. As a workaround I have added the following:

```yml
- command: gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)
```

which seems to be failing with the following error:

![image](https://user-images.githubusercontent.com/46004116/95669206-4a553f00-0b97-11eb-8dee-6dccc8894588.png)

**Comment by svenefftinge** [comment]
Seems to be a timing issue that has been resolved in staging already (due to changes in architecture).
Can you try and confirm?

**Comment by nisarhassan12** [comment]
Thanks. I have tested twice the below error is gone
![image](https://user-images.githubusercontent.com/46004116/95733670-e5cfd800-0c9b-11eb-915b-fe1081361069.png)

but now when I open the workspace and once the 
```
- command: gp await-port 8000 && sleep 3 && gp preview $(gp url 8000)
```
has been executed I get the following the preview seems to be taking forever to load and the prompt is asking me to Save some file:

![image](https://user-images.githubusercontent.com/46004116/95734125-87572980-0c9c-11eb-8f90-faa35ec8e52a.png)

trying the Open in Browsers results in the same thing i.e the Save file prompt.

**Comment by svenefftinge** [comment]
closed as dupe of https://github.com/gitpod-io/gitpod/issues/2041
